src/main/java/org/saltyrtc/client/signaling/peers/Initiator.java

```diff
@@ -34,6 +48,40 @@ public String getName() {
         return "Initiator";
     }
 
+    @NonNull
+    @Override
+    public SharedKeyStore getPermanentSharedKey() {
+        //noinspection ConstantConditions: Set in initiator
+        return this.permanentSharedKey;
+    }
+
+    /**
+     * Return the temporary local session key, if set.
+     *
+     * After calling this function, the temporary local session key will be set to null.
+     */
+    @Nullable
+    public KeyStore extractTmpLocalSessionKey() {
+        final KeyStore ks = this.tmpLocalSessionKey;
```

Comment:
Perhaps throw an exception instead of returning `null` in case the key was not set?

Reply:
Done!